lib/core/network/protocols/protocol.js

```diff
@@ -21,47 +21,62 @@
 
 'use strict';
 
-const
-  assert = require('assert'),
-  bytes = require('../../../util/bytes'),
-  http = require('http');
+const assert = require('assert');
+const http = require('http');
+
+const bytes = require('../../../util/bytes');
 
 class Protocol {
-  constructor() {
+  constructor (name) {
     this.maxRequestSize = null;
     this.entryPoint = null;
-    this.name = null;
+    this.name = name;
     this.config = {};
+
+    this.initCalled = false;
+
+    Reflect.defineProperty(this, '_kuzzle', {
+      value: null,
+      writable: true
+    });
+  }
+
+  get kuzzle () {
+    return this._kuzzle;
   }
 
   /**
-   * @param {string} name - Protocol name (used for accessor)
+   * @param {string} name - Protocol name (used for accessor) @deprecated
    * @param {EmbeddedEntryPoint} entryPoint
    *
    * @return {Promise<boolean>}
    */
-  init(name, entryPoint) {
+  async init(name, entryPoint) {
+    this._kuzzle = entryPoint.kuzzle;
     this.entryPoint = entryPoint;
 
-    return Promise.resolve()
-      .then(() => {
-        this.name = name;
-        this.maxRequestSize = bytes(entryPoint.config.maxRequestSize);
+    // name should be passed in the constructor
+    if (! this.name) {
+      this.name = name;
+    }
```

Comment:
(nitpicking) possible candidate for an assertion test, failing init if a `name` value is provided both to the constructor and to the init methods.